src/main/java/com/internship/backend/controller/LocationController.java

```diff
@@ -31,40 +29,37 @@ public class LocationController {
     @PostMapping("/add")
     public ResponseEntity<Location> addLocation(@RequestBody LocationDTO locationDTO){
         try{
-            Location baseLocation = locationService.fromDTO(locationDTO);
-            baseLocation = locationService.addLocation(baseLocation);
-            Log.info("Added location " + baseLocation.getName());
-            return ok(baseLocation);
+            Log.info("Added location " + locationDTO.getName());
+            return ok(locationService.addLocation(locationService.parse(locationDTO)));
         }catch (LocationAlreadyExistsException e)
         {
             Log.error(e.getMessage());
-            return ResponseEntity.status(HttpStatus.CONFLICT).body(null);
+            return status(HttpStatus.CONFLICT).body(null);
         }
     }
 
     @GetMapping("/getAll")
     public ResponseEntity<List<Location>> getAllLocations() {
-        Optional<List<Location>> location = Optional.ofNullable(locationService.getAllLocations());
-        if(location.isPresent()) {
-            Log.info("Get all: ", location);
+        if(ofNullable(locationService.getAllLocations()).isPresent()) {
+            Log.info("Get all: ");
```

Comment:
Could the message of the log be improved? Currently if we get this log we don't know to who is refer to.